utility_analysis_new/histograms.py

```diff
@@ -319,24 +344,169 @@ def compute_dataset_histograms(col, data_extractors: pipeline_dp.DataExtractors,
     # col: (pid, pk)
 
     # Compute histograms.
-    cross_partition_histogram = _compute_l0_contributions_histogram(
+    l0_contributions_histogram = _compute_l0_contributions_histogram(
         col_distinct, backend)
-    per_partition_histogram = _compute_linf_contributions_histogram(
+    linf_contributions_histogram = _compute_linf_contributions_histogram(
         col, backend)
     partition_count_histogram = _compute_partition_count_histogram(col, backend)
     partition_privacy_id_count_histogram = _compute_partition_privacy_id_count_histogram(
         col_distinct, backend)
     # all histograms are 1 element collections which contains ContributionHistogram
 
     # Combine histograms to DatasetHistograms.
-    histograms = backend.flatten(
-        (cross_partition_histogram, per_partition_histogram,
-         partition_count_histogram, partition_privacy_id_count_histogram),
-        "Histograms to one collection")
-    # histograms: 4 elements collection with elements ContributionHistogram
+    return _to_dataset_histograms([
+        l0_contributions_histogram, linf_contributions_histogram,
+        partition_count_histogram, partition_privacy_id_count_histogram
+    ], backend)
 
-    histograms = backend.to_list(histograms, "Histograms to List")
-    # 1 element collection: [ContributionHistogram]
-    return backend.map(histograms, _list_to_contribution_histograms,
-                       "To ContributionHistograms")
-    # 1 element collection: (DatasetHistograms)
+
+########## Computing histograms on pre-aggregated datasets ####################
+# More details on pre-aggregate datatests are in docstring of function
+# pre_aggregation.preaggregate.
+
+
+def _compute_l0_contributions_histogram_on_preaggregted(
+        col, backend: pipeline_backend.PipelineBackend):
+    """Computes histogram of cross partition privacy id contributions.
```

Comment:
nit: opt: I am not sure how easy it is to misinterpret "cross partition privacy id contributions". Would 
"Computes histogram of the number of distinct partitions contributed by a privacy ID" make it more explicit?

Reply:
Thanks, I like that.